### PR TITLE
docs: update EntityManager#transactional to EntityManager#wrapInTransaction

### PR DESCRIPTION
**Pull request by IndraGunawan**
update docs based on this change https://github.com/doctrine/orm/pull/9278/files#diff-434625111321abb225365806085a3753cb854d098ed9b4b6dc2001235c7f7bc4R5

### CODE REVIEW

**Review by greg0ire — DISMISSED** [review]

File: docs/en/reference/transactions-and-concurrency.rst [hunk 1]
```diff
@@ -106,7 +115,7 @@ functionally equivalent to the previously shown code looks as follows:
 
 .. warning::
 
-    For historical reasons, ``EntityManager#transactional($func)`` will return
+    For historical reasons, ``EntityManager#wrapInTransaction($func)`` will return
```
Comment (beberlei) [inline docs/en/reference/transactions-and-concurrency.rst]: Didnt the return value change in the new method? Afair it was introduced to fix these historical reasons.

Comment (IndraGunawan) [reply docs/en/reference/transactions-and-concurrency.rst]: on v2.9 https://github.com/doctrine/orm/blob/2.19.5/src/EntityManager.php#L255
on v3.1 https://github.com/doctrine/orm/blob/3.1.x/src/EntityManager.php#L187

should i remove that `historical reason` ?

Comment (greg0ire) [reply docs/en/reference/transactions-and-concurrency.rst]: Oh right, yeah, I'd say you can remove the entire warning, the difference between both methods is             

```patch
- return $return ?: true;
+ return $return;
```

So this warning is no longer accurate and can be removed.

**Review by SenseException — DISMISSED** [review]

**Comment by derrabus** [comment]
Do we need to fix this on 2.x as well?

**Comment by greg0ire** [comment]
I think so. Rebased.

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @IndraGunawan !

**Pull request merged**